core/trino-main/src/test/java/io/trino/sql/planner/planprinter/TestExpressionAnonymization.java

```diff
@@ -0,0 +1,111 @@
+ * limitations under the License.
+ */
+package io.trino.sql.planner.planprinter;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.sql.tree.BinaryLiteral;
+import io.trino.sql.tree.BooleanLiteral;
+import io.trino.sql.tree.CharLiteral;
+import io.trino.sql.tree.ComparisonExpression;
+import io.trino.sql.tree.DecimalLiteral;
+import io.trino.sql.tree.DoubleLiteral;
+import io.trino.sql.tree.GenericLiteral;
+import io.trino.sql.tree.IntervalLiteral;
+import io.trino.sql.tree.Literal;
+import io.trino.sql.tree.LogicalExpression;
+import io.trino.sql.tree.LongLiteral;
+import io.trino.sql.tree.NodeLocation;
+import io.trino.sql.tree.NullLiteral;
+import io.trino.sql.tree.StringLiteral;
+import io.trino.sql.tree.SymbolReference;
+import io.trino.sql.tree.TimeLiteral;
+import io.trino.sql.tree.TimestampLiteral;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.trino.sql.tree.ComparisonExpression.Operator.EQUAL;
+import static io.trino.sql.tree.ComparisonExpression.Operator.GREATER_THAN;
+import static io.trino.sql.tree.ComparisonExpression.Operator.LESS_THAN;
+import static io.trino.sql.tree.LogicalExpression.Operator.AND;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestExpressionAnonymization
+{
+    @Test
+    public void testSymbolReferenceAnonymization()
+    {
+        LogicalExpression expression = new LogicalExpression(AND, ImmutableList.of(
+                new ComparisonExpression(GREATER_THAN, new SymbolReference("a"), new LongLiteral("1")),
+                new ComparisonExpression(LESS_THAN, new SymbolReference("b"), new LongLiteral("2")),
+                new ComparisonExpression(EQUAL, new SymbolReference("c"), new LongLiteral("3"))));
+        HashingAnonymizer anonymizer = new HashingAnonymizer("test_environment");
+        assertThat(anonymizer.anonymize(expression))
+                .isEqualTo("((\"symbol_8580e4f2b35a15b4ebf8aa4f811f1088\" > 1) AND (\"symbol_eeb8ac2f5312455a9c9e764414238adf\" < 1) AND (\"symbol_cdc2b5337778e3defcfb85ad2dddfdf1\" = 1))");
+    }
+
+    @Test(dataProvider = "literals")
+    public void testLiteralAnonymization(Literal actual, Literal expected)
+    {
+        HashingAnonymizer anonymizer = new HashingAnonymizer("test_environment");
+        assertThat(anonymizer.anonymize(actual)).isEqualTo(expected.toString());
+    }
+
+    @DataProvider
+    public static Object[][] literals()
+    {
+        return new Object[][] {
+                {new BinaryLiteral("DEF321"), new BinaryLiteral("ABC123")},
+                {new StringLiteral("abc"), new StringLiteral("literal_f3e2a5a815a19f19e4dd00683d32bc64")},
+                {new GenericLiteral("bigint", "1"), new GenericLiteral("bigint", "literal_5be854d593624a18f2c4cf6a1f08ba9c")},
+                {new CharLiteral("a"), new CharLiteral("literal_8580e4f2b35a15b4ebf8aa4f811f1088")},
```

Comment:
This is misleading. The "type" of the anonymized literal does not match the type of the original literal.

Reply:
On a side note, anonymizing literals is very tricky, though. If we want the expressions to continue to be semantically valid and equivalent even after literal anonymization, the process needs to be aware of the relationship between all the expressions and constraints imposed. For example, if the expression were `age > 30 AND age < 50`, replacing it with something like `a > 100 AND a < 20` would be non-sensical.

IMO, the anonymization process should behave as a sort of redaction process. The resulting expressions are no necessarily syntactically or semantically equivalent as the original ones, but it's also clear and explicit where values have been redacted. For example the result of anonymizing the expression above might be `a > [redacted] AND a < [redacted]`.

On this point, any tests that relies on building syntax trees from the anonymized representation is potentially broken since the anonymized/redacted expressions are not necessarily valid SQL expressions. Therefore, any assertions should be made on the final representation.

Reply:
>  replacing it with something like a > 100 AND a < 20 would be non-sensical.

I agree with this.

> Therefore, any assertions should be made on the final representation.

Anyways we are only doing anonymization as part of plan printing so I doubt we are going to use anonymized/redacted expressions to build a syntax tree.

I'm thinking if we can represent all literals in this format, it will help us uniquely identify it as part of the anonymized plan while maintaining consistency.
`new StringLiteral("literal_<type>_<hashvalue>")` or `new GenericLiteral("<type>", "<hashvalue>")`

WDYT?

Reply:
Since the resulting anonymized expressions are not proper SQL, we should not be constructing objects from the SQL AST with their values. Instead, the tests should assert that when anonymizing, say `new StringLiteral("abc")` we get a String with value `literal_f3e2a5a815a19f19e4dd00683d32bc64` (or a more concise value by doing the mapping as I described elsewhere), not a `StringLiteral`.

Reply:
> because we are using the murmur3 hashing mechanism, the mapping is one-to-one

While collision rate might low, there's no way to guarantee a one-to-one mapping. 

However, that wasn't the point of my comment -- it was about making the names as concise as possible to make the anonymized plan more "readable" and less verbose.

Reply:
@martint If the current names look okay to you. Can we merge this PR?